File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.py
class Solution(object):
    def countPartitions(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        total = sum(nums)
        currSum = nums[0]
        partitions = 0
        for i in range(1, len(nums)):
            if ((total - 2 * currSum) % 2 == 0):
                partitions += 1
        return partitions

File: 3704-count-partitions-with-even-sum-difference/count-partitions-with-even-sum-difference.java
class Solution {
    public int countPartitions(int[] nums) {
        int sum = Arrays.stream(nums).sum();
        int currentSum = nums[0];
        int partitions = 0;
        for (int i = 1; i < nums.length; i++) {
            if ((sum - 2 * currentSum) % 2 == 0) {
                partitions++;
            }
            currentSum += nums[i];
        }
        return partitions;
    }
}